### PR TITLE
python312Packages.nodriver: init at 0.34

### PR DESCRIPTION
**Pull request by liam-murphy14**
## Description of changes

Add python package for nodriver. homepage: https://github.com/ultrafunkamsterdam/nodriver

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/python-packages.nix
Comment (Aleksanaa) [inline pkgs/top-level/python-packages.nix:1254]: Why's this line?

File: pkgs/development/python-modules/nodriver/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/nodriver/default.nix:43]:
```suggestion
    longDescription = ''
```

File: pkgs/development/python-modules/nodriver/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/nodriver/default.nix:44]:
```suggestion
      Successor of Undetected-Chromedriver. Providing a blazing fast framework for web
      automation, webscraping, bots and any other creative ideas which are normally
      hindered by annoying anti bot systems like Captcha / CloudFlare / Imperva / hCaptcha
```

File: pkgs/development/python-modules/nodriver/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/nodriver/default.nix:42]:
```suggestion
    license = licenses.agpl3Only;
    description = "Web automation framework which can bypass bot detection";
```

File: pkgs/development/python-modules/nodriver/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/nodriver/default.nix:14]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/nodriver/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/nodriver/default.nix:39]: better to avoid `with lib`

**Comment by ToasterUwU** [comment]
Result of `nixpkgs-review pr 326242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nodriver</li>
    <li>python311Packages.nodriver.dist</li>
    <li>python312Packages.nodriver</li>
    <li>python312Packages.nodriver.dist</li>
  </ul>
</details>

**Comment by ToasterUwU** [comment]
Builds, and also works. Just tested it with the more complex example from their README, and there were no issues.

So as soon as this has the changes suggested by @Aleksanaa, this has my approval.

File: pkgs/development/python-modules/nodriver/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/nodriver/default.nix:38]:
```suggestion
  # no tests
  doCheck = false;
```
It is recommended to explicitly disable tests in python modules.

> Make sure the tests are enabled using for example [pytestCheckHook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook) and, in the case of libraries, are passing for all interpreters. If certain tests fail they can be disabled individually. Try to avoid disabling the tests altogether. In any case, when you disable tests, leave a comment explaining why.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

**Review by natsukium — COMMENTED** [review]
Could you format with `nixfmt-rfc-style`?

File: pkgs/top-level/python-packages.nix
Comment (liam-murphy14) [reply pkgs/top-level/python-packages.nix:1254]: i didnt add this i don't think ? local `git diff master` doesn't show it..... if it is showing up here tho I can remove it

**Comment by liam-murphy14** [comment]
Hey @natsukium @Aleksanaa @ToasterUwU thanks for looking at this package. I will make the requested changes later today !

File: pkgs/top-level/python-packages.nix
Comment (liam-murphy14) [reply pkgs/top-level/python-packages.nix]: removed it in latest push

**Comment by liam-murphy14** [comment]
Hey @Aleksanaa and @natsukium, updated with all your suggestions, thanks again !! Let me know if you have any more thoughts.

File: pkgs/development/python-modules/nodriver/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/nodriver/default.nix:40]:
```suggestion
```
It seems that the upstream has no changelog.

Besides this, lgtm

Comment (ToasterUwU) [reply pkgs/development/python-modules/nodriver/default.nix:40]: Yeah seems like it. Not sure if this is relevant, but there seems to be a somewhat hidden documentation page: https://ultrafunkamsterdam.github.io/nodriver/index.html

Not sure if there is a meta key for "Documentation" or anything.

Comment (liam-murphy14) [reply pkgs/development/python-modules/nodriver/default.nix]: removed the changelog, the documentation is accessible from github homepage so should be fine, i didnt see a meta attribute for "Documentation" unfortunately

**Review by ToasterUwU — APPROVED** [review]

**Review by ToasterUwU — APPROVED** [review]

**Comment by ToasterUwU** [comment]
@liam-murphy14 I think you should rerequest reviews now that you fixed the things they wanted changed, that way they will see it and also approve

**Comment by ToasterUwU** [comment]
Result of `nixpkgs-review pr 326242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nodriver</li>
    <li>python311Packages.nodriver.dist</li>
    <li>python312Packages.nodriver</li>
    <li>python312Packages.nodriver.dist</li>
  </ul>
</details>

**Review by ToasterUwU — APPROVED** [review]

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 326242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python311Packages.nodriver</li>
<li>python311Packages.nodriver.dist</li>
<li>python312Packages.nodriver</li>
<li>python312Packages.nodriver.dist</li>
</ul>
</details>

**Comment by ToasterUwU** [comment]
Result of `nixpkgs-review pr 326242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nodriver</li>
    <li>python311Packages.nodriver.dist</li>
    <li>python312Packages.nodriver</li>
    <li>python312Packages.nodriver.dist</li>
  </ul>
</details>

**Review by ToasterUwU — APPROVED** [review]

**Comment by ToasterUwU** [comment]
@natsukium @Aleksanaa Sorry for the ping, if you dont have time atm, just ignore the rest of the message.

I just want to send a little reminder that the changes you requested have been implemented and that we are waiting for your reviews. Both of you seem to have merge permissions, so it would be very nice if we could get this merged any time soon now that all issues are resolved.
Sorry for the explicit reminder and ping, im just not sure you saw the review requests.

**Comment by SuperSandro2000** [comment]
@ofborg build python312Packages.nodriver

**Pull request merged**
